### DISCUSSION TITLE
Machine Learning Function Collapsed

### DISCUSSION THREAD

**Discussion by isZYKerman** [immich-app/immich#13398] [Q&A] [answered]
I run immich (server, ver.1.117) on TrueNAS. I am a Chinese user so I want to change the machine learning model. I downloaded the model from huggyface and manually copied it to /mlcache/clip folder in the machine learning container. However after I changed the model selection in the immich settings panel, the search function becomes unavailable, with a 'sever error' message popping up every time. I tried restarting the whole app but the error persists. When I went back to the /mlcache/clip folder, I find that it is emptied every time when I restart the app. What can I do to solve the problem?

**Comment by mertalev** [comment, marked as answer]
Could you share the logs for machine learning? Also, can you describe how you downloaded and moved the model? It expects a specific folder structure that also includes the JSON files.

For reference, this should be a correct way to download the model manually:

```bash
pip install huggingface_hub[cli]
export MODEL_NAME="XLM-Roberta-Large-Vit-B-16Plus"
huggingface-cli download immich-app/$MODEL_NAME --local-dir $MODEL_NAME --cache-dir $MODEL_NAME
docker cp $MODEL_NAME immich_machine_learning:/cache/clip
```

**Comment by isZYKerman** [comment]
Here is the log after I started the app and click the "All" button for "Smart Search" in the administration panel.
[log.txt](https://github.com/user-attachments/files/17352961/log.txt)

-----------------------------------------------------------------------------------------------

As for how I downloaded and moved the model, I downloaded the model manually by hitting the download button here:
![image](https://github.com/user-attachments/assets/f36daa51-a63c-43d0-8fc4-15402b78faae)
Then I organized it into the form of

XLM-Roberta-Large-Vit-B-16Plus
|
|--textual
|    |
|    |- [a bunch files downloaded from the website]
|
|--visual
|    |
|    |-model.onnx
|    |-visual_preprocess_cfg.json
|
|--config.json
|--.gitattributes
|--README.md

Then I copied the whole XLM-Roberta-Large-Vit-B-16Plus folder to /mnt/Data/TANK/TEMP on my NAS via SMB. After that, I went to the shell on my NAS and executed

`
docker ps /mnt/Data/TANK/TEMP/XLM-Roberta-Large-Vit-B-16Plus ix-immich-machine-learning-1:/mlcache/clip/XLM-Roberta-Large-Vit-B-16Plus
`

And I went back to Immich to change the model type in the administration. The Smart Search function collapsed then and I tried to restart the whole app. After the re-deployment the function is still not working. I retried a few times but little use. I emptied the folder in one of my tries. After that, the /mlcache folder is emptied every time I restarted the app.

-----------------------------------------------------------------------------------------------

I'll try the correct method you provided. I wish this solved the problem.

**Reply by isZYKerman** [reply]
Your commands work perfectly. Thank you for helping me with this stuff.
By the way, I recommend that the instructions above should be part of the document of how to change ML models, as for the document now only provides unspecific descriptions and a link to huggingface.
